navx/registerio.py

```diff
@@ -192,7 +192,11 @@ def getCurrentData(self):
         else:
             read_count = IMURegisters.NAVX_REG_QUAT_OFFSET_Z_H + 1 - first_address
 
-        curr_data = self.io_provider.read(first_address, read_count)
+        try:
+            curr_data = self.io_provider.read(first_address, read_count)
+        except IOError as e:
+            logger.warning("Error reading data, ignoring: %s", e)
```

Comment:
This should probably be rate limited.

Reply:
Probably a good idea, although the Java libraries don't rate limit either when logging is enabled. (See my comment below though.)